[Workers] Wrangler: Document keep_vars, and how wrangler.toml is a source of truth (#5941)
cloudflare/cloudflare-docs: content/workers/wrangler/configuration.md

# Configure `wrangler.toml`

> Wrangler optionally uses a `wrangler.toml` configuration file to customize the development and publishing setup for a Worker. By default, `wrangler.toml` is the [source of truth](#source-of-truth) for configuring a Worker.

Reviewer: I am not sure this is strictly true. Perhaps we could relax the statement a bit to:

> It is best practice to treat `wrangler.toml` as the [source of truth](#source-of-truth) for configuring a Worker.

# The same as routes, but only one.
route = "routes" | { pattern = "*", zone_id = "ZONE_ID" } | { pattern = "*", zone_name = "ZONE_NAME" }

# Stop wrangler from deleting vars that are not present in the wrangler.toml
# By default Wrangler will remove all vars and replace them with those found in the wrangler.toml configuration.
# If your development approach is to modify vars after deployment via the dashboard you may wish to enable this option.
keep_vars = true

# A map of environment variables to set when deploying your Worker.
# @default `{}`
# not inherited
[vars]
KEY = "value"

> # Lets you call Workers periodically, much like a cron job.

Reviewer: Not strictly relevant to this PR but I would say:

```suggestion
# Lets you trigger Workers periodically, much like a cron job.
```

---
SECRET_KEY = "value"
```

## Source of truth

> By default, `wrangler.toml` serves as a source of truth for your Worker configuration. This allows you to treat `wrangler.toml` as a form of Infrastructure as Code.

Reviewer: Again, I am not sure we can say this. Strictly the source of truth is whatever is currently deployed on EWC. Developers can, and often do, make changes to their Worker via the dashboard (or even via custom scripts that hit EWC directly). Moreover, secrets (encrypted vars) cannot be specified at all in wrangler.toml right now.

I think we should say that it is recommended practice to use wrangler.toml as the source of truth and avoid making changes via other methods if possible.

Author: Right, I see what you mean

> It is a recommended best-practice to treat `wrangler.toml` as a source of truth for your Worker configuration, and avoid making changes via the Cloudflare dashboard. This allows you to treat `wrangler.toml` as a form of Infrastructure as Code.

Reviewer 2: ```suggestion
It is a recommended best practice to treat `wrangler.toml` as a source of truth for your Worker configuration, and avoid making changes via the Cloudflare dashboard. This allows you to treat `wrangler.toml` as a form of Infrastructure as Code.
```

> If you change your environment variables in the Cloudflare dashboard, Wrangler _will_ override them the next time you deploy. If you want to disable this behavior, add `keep_vars = true` to your `wrangler.toml`. 

Reviewer 2: ```suggestion
If you change your environment variables in the Cloudflare dashboard, Wrangler will override them the next time you deploy. If you want to disable this behavior, add `keep_vars = true` to your `wrangler.toml`. 
```

> Note that wrangler won't delete your secrets (encrypted environment variables) unless you run `wrangler secret delete <key>`.

Reviewer 2: ```suggestion
Note that Wrangler will not delete your secrets (encrypted environment variables) unless you run `wrangler secret delete <key>`.
```
